oz_tree_build/newick/extract_trees.py

```diff
@@ -24,86 +21,85 @@
             for subtree in subtrees:
-                if subtree["ancestors_needed"] and node_start_index < subtree["start"]:
+                if subtree["ancestors_needed"] and node_start_idx < subtree["start"]:
                     subtree["tree_string"] = "(" + subtree["tree_string"] + ")" + taxon
                     subtree["ancestors_needed"] -= 1
                     overall_ancestor_needed -= 1
 
         # If this taxon or ott is in the excluded list, add it to the excluded ranges
         if taxon in excluded_taxa or ott in excluded_taxa:
             # Use different logic depending on comma position
-            if newick_tree[node_start_index - 1] == ",":
-                # Exclude the comma before the excluded taxon. e.g. (A,B,REMOVE_ME) --> (A,B)
-                excluded_range = (node_start_index - 1, node_end_index)
-            elif newick_tree[node_end_index] == ",":
-                # Exclude the comma after the excluded taxon. e.g. (REMOVE_ME,B,C) --> (B,C)
-                excluded_range = (node_start_index, node_end_index + 1)
+            if newick_tree[node_start_idx - 1] == ",":
+                # Exclude comma before the excluded taxon. e.g. (A,B,REMOVE_ME) --> (A,B)
+                excluded_range = (node_start_idx - 1, node_end_idx)
+            elif newick_tree[node_end_idx] == ",":
+                # Exclude  comma after the excluded taxon. e.g. (REMOVE_ME,B,C) --> (B,C)
+                excluded_range = (node_start_idx, node_end_idx + 1)
             else:
                 # Otherwise just exclude the taxon, e.g. (REMOVE_ME) --> ()
                 # This can lead to empty brackets, but that's harmless enough
-                excluded_range = (node_start_index, node_end_index)
+                excluded_range = (node_start_idx, node_end_idx)
             excluded_ranges.append(excluded_range)
-            # Sort the excluded ranges by start index. Not efficient, but not on critical path
+            # Sort excluded ranges by start index. Not efficient but not on critical path
             excluded_ranges.sort(key=lambda x: x[0])
 
         # If this taxon or ott is in the target list, add it to the nodes list
         if taxon in target_taxa or ott in target_taxa:
             # First, remove it from the target list
             target_taxa.remove(taxon if taxon in target_taxa else ott)
 
-            tree_string = ""
-
-            def string_to_append(start, end):
-                # Fix up situation that would end up generating "(,"
-                if tree_string and tree_string[-1] == "(" and newick_tree[start] == ",":
-                    start += 1
-                return newick_tree[start:end]
+            tree_str = ""
 
             # Extract the subtree for this node, skipping over excluded ranges
-            prev_range = (node_start_index, node_start_index)
-            for range in excluded_ranges:
+            prev_r = (node_start_idx, node_start_idx)
+            for r in excluded_ranges:
```

Comment:
Are these renames like `prev_range -> prev_r` and `range -> r` enforced by Ruff?

Reply:
No, that was me making some changes. Ruff wasn't allowing "range" as a variable name because it conflicts with the python builtin `range` function. So I took the opportunity to shorted the variable name anyway (same with index -> idx). It's often a little easier to read with slightly shorter names, although perhaps `r` is a little too terse.